nmigen/test/test_hdl_ast.py

```diff
@@ -201,6 +201,24 @@ def test_getitem_wrong(self):
                 msg="Cannot index value with 'str'"):
             Const(31)["str"]
 
+    def test_rotate_left(self):
+        self.assertRepr(Value.cast(256).rotate_left(1), "(cat (slice (const 9'd256) 8:9) (slice (const 9'd256) 0:8))")
+        self.assertRepr(Value.cast(256).rotate_left(7), "(cat (slice (const 9'd256) 2:9) (slice (const 9'd256) 0:2))")
+        self.assertRepr(Value.cast(256).rotate_left(-1), "(cat (slice (const 9'd256) 1:9) (slice (const 9'd256) 0:1))")
+        self.assertRepr(Value.cast(256).rotate_left(-7), "(cat (slice (const 9'd256) 7:9) (slice (const 9'd256) 0:7))")
+        with self.assertRaises(TypeError,
```

Comment:
Tests for API misuse should be separate from tests for correct API use (extract these into something like `test_rotate_left_wrong`).